rules-java/api/src/main/java/org/jboss/windup/rules/apps/java/reporting/freemarker/filepath/GetPrettyPathForFile.java

```diff
@@ -44,4 +48,29 @@ public String getPath(JavaSourceFileModel javaSourceModel)
 
         return packageName == null || packageName.equals("") ? filename : packageName + "." + filename;
     }
+
+    public static void addPrettyPathToModel(FileModel fileModel, GraphContext context) {
+        if (fileModel == null) {
+            return;
+        }
+
+        ArrayList<String> types = fileModel.asVertex().getProperty(WindupVertexFrame.TYPE_PROP);
+
+        if (types.contains(JavaClassFileModel.TYPE)) {
+            //if (fileModel instanceof JavaClassFileModel) {
+            JavaClassFileModel jcfm = ((JavaClassFileModel) fileModel);
+            jcfm.setCachedPrettyPath(jcfm.getPrettyPathWithinProject(true));
+            //} else if (fileModel instanceof ReportResourceFileModel) {
+        } else if (types.contains(JavaSourceFileModel.TYPE)) {
+            JavaSourceFileModel jsfm = context.getFramed().frame(fileModel.asVertex(), JavaSourceFileModel.class);
+            // ((JavaSourceFileModel)fileModel);
+
+            jsfm.setCachedPrettyPath(jsfm.getPrettyPathWithinProject(true));
```

Comment:
I have a problem here. I tried type casting, I tried reframe model to `JavaSourceFileModel`, but in certain cases, none of these work and `getPrettyPathWithinProject(true)` is still being called on `FileModel` instead of `JavaSourceFileModel`